holoviews/core/spaces.py

```diff
@@ -749,14 +755,34 @@ def __init__(self, callback, initial_items=None, **params):
                          'and no longer needs to be specified.')
             del params['sampled']
 
-        super(DynamicMap, self).__init__(initial_items, callback=callback, **params)
-        invalid = [s for s in self.streams if not isinstance(s, Stream)]
+        # Validate streams by ensuring parameterized objects and methods are wrapped in ParamStream
```

Comment:
How about a utility here? Something like:

```
valid, invalid = util.parameterized_streams(streams)
```

I'm not quite sure what you need `invalid` for though...

Reply:
If you look below you'll see that invalid streams will raise errors. Happy to factor that out if you want.

Reply:
Oops missed that. I think the utility could raise the exception for invalid inputs.